src/components/MyKiva/BadgeContainer.vue

```diff
@@ -150,12 +158,19 @@ const handleBadgeClick = () => {
 			secondShine.value.$el.classList.add('animate');
 		}, 10);
 	}
+};
 
-	if (isLocked.value) {
-		animateLock.value = true;
-		setTimeout(() => { animateLock.value = false; }, 1600);
+watch(() => props.showShine, newValue => {
+	if (newValue) {
+		handleShine();
 	}
-};
+});
+
+onMounted(() => {
+	if (props.showShine) {
+		handleShine();
```

Comment:
Could this be related with `{ immediate: true }` on the `watch`?

Reply:
Already tested and doesn't work...